### ISSUE TITLE
Saver AIS data from receiver localy

### ISSUE THREAD

**Issue by gabrielspadon** [AISViz/AISdb#117]
I noticed that in many cases, the processes that run remotely (receiving the data from the Raspberry) can fail for some unknown reason. I believe we should save the data locally, even if we don't have much space, just to avoid losing it. If the memory overflows, we clear the data and start storing it again.

**Comment by Jinnkunn** [comment]
Yes, we should proceed with caution, particularly because we’re using an SD card for storage. We also need to determine an optimal buffer size to limit locally stored data. If the stored data exceeds this size, older data should automatically be deleted to maintain storage efficiency.

Alternatively, if we need to store data locally for an extended period, we could add an additional storage device to the Pi to reduce frequent read-write operations on the SD card.

From: Gabriel Spadon ***@***.***>
Date: Friday, November 1, 2024 at 2:49 PM
To: AISViz/AISdb ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [AISViz/AISdb] Saver AIS data from receiver localy (Issue #117)

I noticed that in many cases, the processes that run remotely (receiving the data from the Raspberry) can fail for some unknown reason. I believe we should save the data locally, even if we don't have much space, just to avoid losing it. If the memory overflows, we clear the data and start storing it again.

**Comment by gabrielspadon** [comment]
Implemented on master. The receiver persists locally through start_receiver(sqlite_dbpath=...), and the durability gap discussed here is now closed, since the receiver flushes both message buffers on a timer as well as on shutdown, so a crash or disconnect no longer drops buffered messages. The remaining idea from this thread, a bounded ring buffer that purges the oldest records when storage fills, is a separate enhancement; please open a focused issue if it is still wanted.
